clientv3/client.go

```diff
@@ -153,10 +153,11 @@ func (cred authTokenCredential) GetRequestMetadata(ctx context.Context, s ...str
 
 // Dial establishes a connection for a given endpoint using the client's config
 func (c *Client) Dial(endpoint string) (*grpc.ClientConn, error) {
-	opts := []grpc.DialOption{
+	opts := make([]grpc.DialOption, 0, 5)
```

Comment:
probably not worth doing this one since it's rarely called and would have to be updated if new dial options are added